### PR TITLE
[TRUS-866] Validation for trust details section

### PR DESCRIPTION
**Pull request by manishsw**
validation for 
- start date
- efrbs start date

### CODE REVIEW

File: app/uk/gov/hmrc/trusts/utils/DomainValidator.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import uk.gov.hmrc.trusts.models.Registration
+import uk.gov.hmrc.trusts.services.TrustsValidationError
+import uk.gov.hmrc.trusts.utils.TypeOfTrust._
+
+class DomainValidator(registration : Registration) extends ValidationUtil {
+
+  val EFRBS_VALIDAION_MESSAGE = "Trusts efrbs start date can be provided for Employment Related trust only."
+
+  def trustStartDateIsNotFutureDate : Option[TrustsValidationError] = {
+    isNotFutureDate(registration.details.trust.get.details.startDate,
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/utils/DomainValidator.scala]: Are we safe to do a `.get` here? What is the trust object is None?

File: app/uk/gov/hmrc/trusts/utils/DomainValidator.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "/details/trust/details/startDate", "Trusts start date")
+  }
+
+  def validateEfrbsDate : Option[TrustsValidationError] = {
+    val isEfrbsDateDefined = registration.details.trust.get.details.efrbsStartDate.isDefined
+    val isEmploymentRelatedTrust = registration.details.trust.get.details.typeOfTrust==EMPLOYMENT_RELATED_TRUST.toString
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/utils/DomainValidator.scala]: Again with `.get`.?

File: app/uk/gov/hmrc/trusts/utils/ValidationUtil.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  def isNotFutureDate(date : Option[DateTime],path : String, key : String ) : Option[TrustsValidationError]= {
+    if (date.isDefined && isAfterToday(date.get)) {
+      isNotFutureDate(date.get, path, key)
+    } else None
+  }
+
+   def isAfterToday(date :DateTime): Boolean = {
+     date.isAfter(new DateTime().plusDays(1).withTimeAtStartOfDay())
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/utils/ValidationUtil.scala]: Isn't this not checking tomorrows day? Or are you checking if it's after midnight of the current day?

File: test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala [hunk 4]
```diff
@@ -52,6 +51,13 @@ class ValidationServiceSpec extends BaseSpec {
         validator.validate[ExistingTrustCheckRequest](jsonString).isLeft mustBe true
       }
     }
-  }
+    "return a list of validaton errors " when {
+      "json request is valid but failed in business rules. " in {
+        val jsonString = JsonUtils.getJsonFromFile("trust-business-validation-fail.json")
+        validator.validate[Registration](jsonString).isLeft mustBe true
+        validator.validate[Registration](jsonString).left.toOption.get.size mustBe 3
```
Comment (adamconder) [inline test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala]: @manishsw Are you aware of http://www.scalatest.org/user_guide/other_goodies#eitherValues to test Eithers?

File: test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala [hunk 5]
```diff
@@ -52,6 +51,13 @@ class ValidationServiceSpec extends BaseSpec {
         validator.validate[ExistingTrustCheckRequest](jsonString).isLeft mustBe true
       }
     }
-  }
+    "return a list of validaton errors " when {
+      "json request is valid but failed in business rules. " in {
+        val jsonString = JsonUtils.getJsonFromFile("trust-business-validation-fail.json")
```
Comment (adamconder) [inline test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala]: Why does this fail? Can we be specific about the case that it fails on?

File: app/uk/gov/hmrc/trusts/utils/ValidationUtil.scala (hunk 3 above)
Comment (manishsw) [reply app/uk/gov/hmrc/trusts/utils/ValidationUtil.scala]: yes

File: test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala (hunk 4 above)
Comment (manishsw) [reply test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala]: updated.

File: test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala (hunk 5 above)
Comment (manishsw) [reply test/uk/gov/hmrc/trusts/services/ValidationServiceSpec.scala]: updated heading.

**Pull request merged**